PollBuddy-Server/frontend/src/components/SchoolPicker/SchoolPicker.js

```diff
@@ -19,25 +19,23 @@ export default class SchoolPicker extends Component {
 
   }
 
-  componentDidMount(){
+  async componentDidMount(){
     if (this.props.schoolInfo == null) {
-      fetch(process.env.REACT_APP_BACKEND_URL + "/schools", 
+      const response = await fetch(process.env.REACT_APP_BACKEND_URL + "/schools",
         {
           method: "GET",
           headers: { "Content-Type": "application/json" },//HEADERS LIKE SO ARE NECESSARY for some reason https://stackoverflow.com/questions/39842013/fetch-post-with-body-data-not-working-params-empty
-        }    
-      ).then(response => response.json())
-      // handle response
-        .then(data => {
-          let schools = this.state.schoolInfo.schools;
-          let schoolLinkDict = this.state.schoolInfo.schoolLinkDict;
-          for (var i = 0; i < data.length; i++) {
-            schools.push({ key: i, label: data[i][0] });
-            schoolLinkDict[data[i][0]] = data[i][1];
-          }
-          this.setState({"schoolInfo": {"schools": schools, "schoolLinkDict": schoolLinkDict}}); //missing semicolon
-          this.props.onDoneLoading(this.state.schoolInfo);
-        });
+        }
+      );
+      const data = await response.json();
+      let schools = this.state.schoolInfo.schools;
+      let schoolLinkDict = this.state.schoolInfo.schoolLinkDict;
+      for (var i = 0; i < data.length; i++) {
+        schools.push({ key: i, label: data[i][0] });
+        schoolLinkDict[data[i][0]] = data[i][1];
+      }
+      this.setState({"schoolInfo": {"schools": schools, "schoolLinkDict": schoolLinkDict}}); //missing semicolon
+      this.props.onDoneLoading(this.state.schoolInfo);
```

Comment:
This should be an optional chain because some components don't give this prop.
`this.props.onDoneLoading?.(this.state.schoolInfo);`